modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/ChannelDTOConverter.java

```diff
@@ -43,7 +43,7 @@ public Channel toDTO(DTOConverterContext dtoConverterContext)
 		throws Exception {
 
 		CommerceChannel commerceChannel =
-			_commerceChannelService.getCommerceChannel(
+			_commerceChannelLocalService.getCommerceChannel(
```

Comment:
We need to check that the user has the membership on the site related to the Channel.

So, if `commerceChannel.getType()` is site type, we check that the user has the membership on `commerceChannel.getSiteGroupId()`